Discovery/src/FortuneTeller/FortuneTellerService/Program.cs

```diff
@@ -22,17 +24,20 @@
 builder.Services.AddDbContext<FortuneDbContext>(options => options.UseInMemoryDatabase("Fortunes"));
 builder.Services.AddHostedService<DatabaseSeederHostedService>();
 
-// Steeltoe: Read credentials to Eureka server from VCAP_SERVICES on CloudFoundry.
-builder.AddCloudFoundryConfiguration();
-builder.Configuration.AddCloudFoundryServiceBindings();
-
 // Steeltoe: Add service discovery clients for Consul and Eureka.
 builder.Services.AddConsulDiscoveryClient();
 builder.Services.AddEurekaDiscoveryClient();
 
 // Steeltoe: Add Config Server to demonstrate discovery-first (resolves the URL to Config Server from Eureka).
 builder.Configuration.AddConfigServer();
 
+if (Platform.IsCloudFoundry)
+{
+    // Steeltoe: Read credentials to Eureka server from VCAP_SERVICES on CloudFoundry.
+    builder.AddCloudFoundryConfiguration();
+    builder.Configuration.AddCloudFoundryServiceBindings();
+}
```

Comment:
This is getting tricky (because the order of service registration matters, which shouldn't be the case). The change in execution order here is needed because `AddEurekaDiscoveryClient()` registers `DefaultApplicationInstanceInfo`. This needs to happen *before* `AddCloudFoundryConfiguration()` executes, otherwise the dev-local service name will become `unknown`.

There are several ways to address this, which we should discuss...

Reply:
Yeah, one of the other samples (Redis DataProtection maybe?) has a code branch using `Platform.IsCloudFoundry` that I'd love to see available more broadly, but it probably doesn't work without either globally requiring the CF config provider or doing some reflection

Reply:
A fix is provided at https://github.com/SteeltoeOSS/Steeltoe/pull/1350. Once approved and merged, I'll revert the changes here and remove the `Platform.IsCloudFoundry` check from the Security/AuthClient sample.